smart_open/smart_open_lib.py

```diff
@@ -945,11 +945,6 @@ def __exit__(self, exc_type, exc_val, exc_tb):
 
 def _patch_pathlib(func):
     """Replace `Path.open` with `func`"""
-    pathlib = sys.modules.get("pathlib", None)
```

Comment:
are you sure about it? as I see in head of the file

https://github.com/RaRe-Technologies/smart_open/blob/dfcaf2e7134a220954ab69e3e064d297f47cd76d/smart_open/smart_open_lib.py#L47-L55

i.e. `pathlib` can be not defined

Reply:
`pathlib2` (for py2?) installed only in test env 

https://github.com/RaRe-Technologies/smart_open/blob/90d4f72065c28ef9bd462a18f6d86950e1d7fdaf/setup.py#L42-L45

so, in case if person use `py2` and install smart_open from pypi (with "normal" mode),  `pathlib` can be undefined. For `py3.5+` no issues here (batteries inside).